charmtools/build/tactics.py

```diff
@@ -353,11 +387,34 @@ def trigger(cls, relpath):
         return relpath in cls.FILENAMES
 
     def read(self):
-        self._raw_data = self.load(self.entity.open())
+        if not self._read:
+            super(LayerYAML, self).read()
+            self.schema['properties'] = {
+                self.current.name: {
+                    'type': 'object',
+                    'properties': self.data.pop('defines', {}),
+                    'default': {},
+                },
+            }
+
+    def combine(self, existing):
+        super(LayerYAML, self).combine(existing)
+        self.schema = utils.deepmerge(existing.schema, self.schema)
+        return self
+
+    def lint(self):
+        if 'options' not in self.data:
+            return True
+        validator = extend_with_default(jsonschema.Draft4Validator)(self.schema)
```

Comment:
@bcsaller I agree that it's not ideal to do modification of the target values during validation, but that's unfortunately what the jsonschema library [recommended](https://python-jsonschema.readthedocs.org/en/latest/faq/#why-doesn-t-my-schema-that-has-a-default-property-actually-set-the-default-on-my-instance).  Ideally, the default setting behavior would be handled just after `combine` but I'm not clear on how to implement that without a lot of manual processing of the schema.